### PR TITLE
add distribution unit test parameter forward-backward

### PR DESCRIPTION
**Pull request by armaan-abraham**
- Remove scaling factor. If we indeed believe that we need a linear degree of freedom for all Lbound, we should instead just find a distribution for Lbound that works and then normalize Lbound before optimization to match this distribution. This leads to much better performance.
- Scale the starting distribution of Rtot based on the distribution of Lbound. 
- Replace the custom newton-raphson method for root finding with scipy.optimize.newton. I found this performs better on datasets which may have numerical instability due to large starting distributions.

### CODE REVIEW

File: maserol/core.py [hunk 1]
```diff
@@ -179,18 +177,25 @@ def optimizeLoss(
     jac_sparsity[*idx, *idx] = rcp_ab_block
     jac_sparsity = np.moveaxis(jac_sparsity, (4, 5), (1, 4))
     jac_sparsity = jac_sparsity.reshape(n_samp * n_rcp * n_ag, n_samp * n_ab * n_ag)
-    jac_sparsity = np.pad(jac_sparsity, ((0, 0), (0, 1)), constant_values=1) # cheeky scaling factor
     jac_sparsity = coo_matrix(jac_sparsity)
 
     print("")
+    inner_opt_kwargs = inner_opt_kwargs or {}
+    default_inner_opt_kwargs = {
```
Comment (aarmey) [inline maserol/core.py:184]: Do you need `default_inner_opt_kwargs` long-term? If not I'd remove.

File: maserol/core.py [hunk 2]
```diff
@@ -136,9 +135,7 @@ def modelLoss(log_x: np.ndarray, cube: Union[xr.DataArray, np.ndarray], Ka: np.n
     if isinstance(cube, xr.DataArray):
         cube = np.array(cube)
     Lbound = inferLbound(cube, *(params + ([] if fitKa else [Ka])), L0=L0, KxStar=KxStar, FcIdx=FcIdx)
-    scaling_factor = log_x[-1]
```
Comment (aarmey) [inline maserol/core.py:139]: Reasoning makes sense for this.

File: maserol/core.py [hunk 3]
```diff
@@ -50,21 +54,16 @@ def custom_root(Rtot: np.ndarray, L0: float, KxStar: float, Ka: np.ndarray, f: i
...
-        df = f1.imag / 1e-9
-        fNew = f0 - f1.real / df
-
-        f0 = phi(fNew, KaRT, fLKa, f)
-
-        if np.linalg.norm(f1.real) < 1e-8:
-            break
-
-    return np.maximum(f0, 0)
+    root = newton(phi_res, f0, maxiter=10000, args=(KaRT, fLKa, f))
```
Comment (aarmey) [inline maserol/core.py:60]: Makes sense. I like that this is simpler, too.

**Review by aarmey — APPROVED** [review]
This is great. See my one question before merging.

File: maserol/core.py (hunk 1 above)
Comment (armaan-abraham) [reply maserol/core.py]: Probably not. Removed.

**Pull request merged**